src/url.js

```diff
@@ -348,6 +351,23 @@ export function isProxyOrigin(url) {
   return urls.cdnProxyRegex.test(url.origin);
 }
 
+/**
+ * Returns whether the URL corresponds to an advertisement with a proxy origin.
+ * @param {string|!Location} url URL of an AMP document.
+ * @return {boolean}
+ */
+export function isAlpProxyOrigin(url) {
```

Comment:
it can be useful if you make it a generic function that returns 'a'. returns null if not on proxy origin.

```
getProxyServingType()
```